app/services/discussion_service.rb

```diff
@@ -1,11 +1,17 @@
 class DiscussionService
   def self.recount_everything!
     # I'm not sure anyone will need this .. but it's cool sql
+    # WHOA I TOTALLY NEEDED IT!
     ActiveRecord::Base.connection.execute(
       "UPDATE discussion_readers SET
-       read_comments_count = (SELECT count(id) FROM comments WHERE discussion_id = discussion_readers.discussion_id AND comments.created_at <= discussion_readers.last_read_at ),
-       read_items_count = (SELECT count(id) FROM events WHERE discussion_id = discussion_readers.discussion_id AND events.created_at <= discussion_readers.last_read_at ),
-       read_salient_items_count = (SELECT count(id) FROM events WHERE discussion_id = discussion_readers.discussion_id AND events.created_at <= discussion_readers.last_read_at AND events.kind IN ('#{Discussion::SALIENT_ITEM_KINDS.join('\', \'')}') )", )
+       read_comments_count = (SELECT count(id) FROM events WHERE discussion_id = discussion_readers.discussion_id AND events.created_at <= discussion_readers.last_read_at AND events.kind = 'new_comment'),
+       read_items_count = (SELECT count(id) FROM events WHERE discussion_id = discussion_readers.discussion_id AND events.created_at <= discussion_readers.last_read_at AND events.kind IN ('#{Discussion::THREAD_ITEM_KINDS.join('\', \'')}') ),
+       read_salient_items_count = (SELECT count(id) FROM events WHERE discussion_id = discussion_readers.discussion_id AND events.created_at <= discussion_readers.last_read_at AND events.kind IN ('#{Discussion::SALIENT_ITEM_KINDS.join('\', \'')}') )")
+    ActiveRecord::Base.connection.execute(
```

Comment:
These updates run pretty slowly; ~13min for 90k discussions... but I think we need em in order to sync up the existing threads, I'm hesistant to pick out a subset of discussions or readers to update, and I don't know of a faster way to run a global update like this.